### DISCUSSION TITLE
stroke <-> vici connections possible?

### DISCUSSION THREAD

**Discussion by JumboJa** [strongswan/strongswan#1389] [General]
Hi

Getting "no authorities/pools/connections found, 0 unloaded", which is _swanctl_ interface, as I understand.
Before I switch to it I'd like to establish the simplest connection using ipsec.
Would anyone help me to understand please, what could be the most reason for failure below? TNX

Wed Nov  2 11:41:05 2022 authpriv.info ipsec_starter[9626]: Starting strongSwan 5.9.5 IPsec [starter]...
Wed Nov  2 11:41:05 2022 authpriv.info ipsec_starter[9626]: no files found matching '/var/ipsec/ipsec.conf'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.5, Linux 5.10.138, mips)
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG]   loaded ca certificate "OU=my.loc" from '/etc/ipsec.d/cacerts/CA_intermediary.cer'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading crls from '/etc/ipsec.d/crls'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] loading secrets from '/etc/ipsec.secrets'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG] expanding file expression '/var/ipsec/ipsec.secrets' failed
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[CFG]   loaded ED25519 private key from '/etc/ipsec.d/private/moonKey.pem'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[LIB] loaded plugins: charon aes sha2 sha1 random nonce x509 pubkey pkcs1 pkcs8 pem gmp curve25519 xcbc hmac kernel-netlink socket-default stroke vici updown
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[JOB] spawning 16 worker threads
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[DMN] executing start script 'load-all' (/usr/sbin/swanctl --load-all --noprompt)
Wed Nov  2 11:41:05 2022 authpriv.info ipsec_starter[9637]: charon (9638) started after 60 ms
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 06[CFG] received stroke: add connection 'net-net'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 06[CFG]   loaded certificate "O=my, CN=95.13x.xx.xx" from 'hex-o.cer'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 06[CFG] added configuration 'net-net'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 04[CFG] received stroke: initiate 'net-net'
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 04[IKE] initiating IKE_SA net-net[1] to 69.11x.xxx.x
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 04[ENC] generating IKE_SA_INIT request 0 [ SA KE No N(NATD_S_IP) N(NATD_D_IP) N(FRAG_SUP) N(HASH_ALG) N(REDIR_SUP) ]
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 04[NET] sending packet: from 192.168.1.3[500] to 69.11x.xxx.x[500] (420 bytes)
**Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[DMN] load-all: no authorities found, 0 unloaded
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[DMN] load-all: no pools found, 0 unloaded
Wed Nov  2 11:41:05 2022 daemon.info ipsec: 00[DMN] load-all: no connections found, 0 unloaded** - **IS THIS THE REASON? OR STROKE STILL WORKS?**
Wed Nov  2 11:41:09 2022 daemon.info ipsec: 09[IKE] retransmit 1 of request with message ID 0
Wed Nov  2 11:41:09 2022 daemon.info ipsec: 09[NET] sending packet: from 192.168.1.3[500] to 69.11x.xxx.x[500] (420 bytes)
Wed Nov  2 11:41:16 2022 daemon.info ipsec: 14[IKE] retransmit 2 of request with message ID 0
Wed Nov  2 11:41:16 2022 daemon.info ipsec: 14[NET] sending packet: from 192.168.1.3[500] to 69.11x.xxx.x[500] (420 bytes)

I found this advice:
_add the following line to your rc.conf file to use the stroke configuration.   strongswan_interface="stroke"_
Is this to force stroke and disable vici?
But there's no /etc/rc.conf  file, so where to put it? (if I need it at all)

**Comment by tobiasbrunner** [comment]
That log message is completely unrelated to any issues you are having. (I just see retransmits, which could have lots of reasons, wrong remote address or IKE version, firewall issues, network problems etc.)

> I found this advice:
> _add the following line to your rc.conf file to use the stroke configuration. strongswan_interface="stroke"_
> Is this to force stroke and disable vici?

No idea what this setting changes exactly. You'd have to ask your distro provider about this.

Technically, both interfaces may be used concurrently, but it's not something we recommend. So you probably want to stick to one (we obviously recommend using swanctl.conf).  The log messages by swanctl don't matter as the swanctl.conf file and the swanctl sub-directories are apparently empty in your case so there won't be any effect on the ipsec.conf configs whatsoever.

**Reply by JumboJa** [reply]
TNX a lot for your feedback! All clear. I was suspecting ipsec (stroke) and swanctl (vici) do not depend on each other, and can work at a time.

Seems it's a NAT/firewall issue (one router is behind NAT, although it's in DMZ). 
Looks like it doesn't receive SA:   net-net[1]: IKEv2 SPIs: s5c942y3409cb6a_i* **0000000000000000_r**

Right?

**Reply by tobiasbrunner** [reply]
As I said, there could be various reasons why there is no response to the initial IKE message. You should definitely check firewall rules and configs.
